packages/vc-verification/src/resolver/evm-data-aggregator.ts

```diff
@@ -0,0 +1,91 @@
+import {
+  CachedDIDDocument,
+  CachedRoleCredential,
+  CachedRoleDefinition,
+  RoleEIP191JWT,
+} from '../models';
+import { IDataAggregator } from './data-aggregator';
+import { IServiceEndpoint } from '@ew-did-registry/did-resolver-interface';
+import { IRoleDefinitionV2 } from '@energyweb/credential-governance';
+import { VerifiableCredential } from '@ew-did-registry/credentials-interface';
+import type { RoleCredentialSubject } from '@energyweb/credential-governance';
+
+/**
+ * A class to aggreagate data from EVMs
+ */
+export class EVMDataAggregator implements IDataAggregator {
+  private cahedRoleDefinition: CachedRoleDefinition = {};
+  private cachedDIDDocument: CachedDIDDocument = {};
+  private cachedRoleCredential: CachedRoleCredential[] = [];
```

Comment:
Why an array for `cacheRoleCredential`? Seems to me like it would be more intuitive as either
- a two layer nested map, where the first key gets index by DID and second key indexes by role namespace
- at least use a map to index by did and iterate to find the matching role